rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/SonarLintRpcClient.java

```diff
@@ -145,6 +145,9 @@ public interface SonarLintRpcClient {
   @JsonNotification
   void didSynchronizeConfigurationScopes(DidSynchronizeConfigurationScopeParams params);
 
+  /**
+   * @throws org.eclipse.lsp4j.jsonrpc.ResponseErrorException with {@link org.sonarsource.sonarlint.core.rpc.protocol.client.ClientErrorCode#CONNECTION_NOT_FOUND} if the connection doesn't exist on the client side
+   */
   @JsonRequest
   CompletableFuture<GetCredentialsResponse> getCredentials(GetCredentialsParams params);
```

Comment:
Should we explicitly add the ResponseErrorException to the method signature so we don't forget to handle it?
`CompletableFuture<GetCredentialsResponse> getCredentials(GetCredentialsParams params) throws ResponseErrorException;`

Reply:
ResponseErrorException is a RuntimeException, so I fear we will get warnings from the SQ analysis and from the IDE if we do.

Reply:
In fact, any call can possibly throw a ResponseErrorException